### DISCUSSION TITLE
Ryzen AI 9 365 / 370 TDP Control?

### DISCUSSION THREAD

**Discussion by Noisrevid** [seerge/g-helper#3215] [Ideas]
Having a Zenbook S16 on hand, would like to see if you could make a custom build with TDP control for Ryzen AI 9 chips?  Has anyone tested this?   I would imagine since undervolting seems to be working fine on AI 9 chips, TDP Control should be working..  Testing on G-Helper currently has the "slider" for wattage but it's not actually changing anything, performance/wattage remains the same no matter what slider setting you choose (and yes, the check box to enable wattage/power control is enabled).

**Comment by seerge** [comment]
@Noisrevid hello, if under "TDP Control" you mean option to set a custom CPU Power limits - then it's not specific to some CPU model. 

- If your device supports custom power limits on hardware / firmware level and corresponding endpoint is exposed by Asus in the firmware - you will see a power sliders. 
- If it's not supported (and you never had that option in MyASUS / ProArt app / Armoury) - then it can't be added.

**Reply by Noisrevid** [reply]
The sliders show up fine in G-Helper..  but appear to not apply or do anything.     Let me know if you need a log.  And by not applying it lets me save in G-Helper at the specified power but looking at performance comparisons with different power levels it seems the machine is not changing any TDP/power to the CPU via G-helper.    HWinfo confirms power is unchanging during load.

**Reply by seerge** [reply]
If sliders appear - then it should be supported. 

- Did you have an option to set power limits in native Asus software?
- How do you test if power limits work or not ? You need to run come CPU heavy benchmark like Cinebench23, keep CPU boost enabled and monitor actual CPU power consumption HWinfo.

Some devices require custom fan curve to be applied next to power limits, in order to make them work. Try to check that as well.

And yes - please upload a log from `%appdata%\GHelper` here.

**Reply by Noisrevid** [reply]
[log.txt](https://github.com/user-attachments/files/17256319/log.txt)

Log attached, changed from 150watt  down to 5w and no change in behavior.

**Reply by seerge** [reply]
@Noisrevid ok, i think your device simply does not support setting power limits in an "asus way" :)

The reason you see them in the UI is cause there is a fallback option for some old AMD CPUs that uses [UXTU](https://github.com/JamesCJ60/Universal-x86-Tuning-Utility) approach.  But they simply don't work here.

Can you quickly check if latest version of UXTU is able to set limits by any chance for your device ?

**Reply by Noisrevid** [reply]
Hard to say since UXTU has been broken for many people for quite a while now..   It doesn't seem to launch (on my G14 either and older AMD devices). .. opens and immediately closes.   It's alright, I don't think it's supported like you said (yet)..   It might take time for the UXTU people to figure out the new Ryzen AI chips.

**Reply by seerge** [reply]
@Noisrevid actually, i have checked UXTU sources, and there are some updates there.

Try this build with power limits and check if they work now ?
[GHelper.zip](https://github.com/user-attachments/files/17257989/GHelper.zip)

**Reply by seerge** [reply]
@Noisrevid is the build i linked able to set power limits for you ?

**Reply by Noisrevid** [reply]
Sorry @seerge I ended up returning the laptop yesterday and didn't get a chance to try this build.   I may be getting to review the new G16 with AI 9 370 though here shortly - so if I do I can test this.

**Reply by seerge** [reply]
@Noisrevid that's a pity. G16 supports power limits in a normal way (via asus endpoints). So there they 100% work :)
What I wanted to test if UXTU's approach is able to set limits in situations where laptop by default don't offer such option.

**Comment by mehtabmahir** [comment]
Hi @seerge, I have the s16 with the ryzen ai 9 365, and this didn't seem to work. I set it to 5 watts, and it didn't seem to slow down at all, so I don't think it works.

**Reply by seerge** [reply]
What is "s16" ? Does your device support power limits via asus apps ?

**Reply by mehtabmahir** [reply]
The Zenbook s16 which you guys were talking about in this thread. It doesn't support that through MyASUS pretty sure.

**Reply by mehtabmahir** [reply]
@seerge I'm talking about the build you shared in this thread. Also, I want to add that there is an issue where it thinks the dedicated gpu is active (it does not exist). If there is any way I can help adding support for this device let me know. I was using this on my g14 and it was working great.

**Reply by seerge** [reply]
@mehtabmahir correct way of testing power limits - would be set all 3 slides to same value , run CPU benchmark and monitor power consumption using HWinfo or similar. 

Do power limits in UXTU or ryzenadj work for you ?

**Reply by mehtabmahir** [reply]
@seerge Yeah set all the sliders (there's only 2 on g helper for this device). I noticed the fan profiles do change but the power limits do not change when set in G Helper. I tried UXTU afterwards, and it does seem to work, now it is boosting higher, and temps go to 95 instead of under 80.

**Reply by seerge** [reply]
@mehtabmahir  please upload a log from `%appdata%\GHelper` here. What does it say when you apply power limits ?

Try to set some fixed limit (like 10w) and run cinebench + check hwinfo's CPU power draw

**Reply by mehtabmahir** [reply]
@seerge After some extensive testing with HWinfo, I can see that the power limits do actually work now. If I set it to 10 watts then it consumes 10 watts, and up to 28 watts if I max it out. I'm not sure why it wasn't working earlier but thank you for your responses. 

Only other issue is that G Helper thinks there is a GPU present even though the system doesn't have one. And one other minor issue is the color temperature settings do not work. I've attached a screenshot to show this.
![image](https://github.com/user-attachments/assets/429476a6-d9e7-4fc8-aabe-6526d50721c4)

**Reply by mehtabmahir** [reply]
Scratch that part about 28 watts, it works great at 40 watts but once it thermal throttles (if you set the power limit too high) it goes to 28 watts so I found 40 watts is the sweetspot for this laptop.

**Reply by mehtabmahir** [reply]
Update: I see that the power limit does get set but after a while it drops down to 22 watts anyways (not thermal throttling)

**Reply by seerge** [reply]
@mehtabmahir most likely something overrides it :) Try to open advanced tab and check `Auto Apply` there.

**Reply by mehtabmahir** [reply]
@seerge Wow thank you so much. I think that fixed it. If there is anything I can do to help fix the rest of the app for this device, please let me know. I haven't coded in C# but I am comfortable with C++

**Reply by mehtabmahir** [reply]
Update: Power limit works for a while but it slowly drops down to 24 watts, its not thermal throttling either. I think it may have something to do with cpu skin temperature.

**Reply by seerge** [reply]
@mehtabmahir probably better to stay under limits that your device supports out of the box.

As for GPU section, i can hide it (as app does for some devices), but then it will also hide line with GPU fan speed. 
Does "GPU" (i.e. second) fan in your system spin with different speeds comparing to a main one ? or both always work in sync ?

**Reply by mehtabmahir** [reply]
@seerge yeah the gpu fan reading is in fact different. But I hope there is a way you can show it without showing the other part. Also it might be better, but asus actually lowered the power limit for this model because people were complaining about heat but I really want to be able to use 30+ watts when gaming. It is nowhere near thermal throttling because of the 24 watt limit, temps are in the 70s

**Reply by mehtabmahir** [reply]
@seerge Maybe you can make it say "GPU Mode: iGPU" instead of "GPU Mode: iGPU + dGPU". That makes more sense and fixes the issue overall. 
One other small issue is when I stop ASUS services, some of the functions do not work like the keyboard backlight turning off with auto brightness, so maybe you can exclude that service? 
Also, the keyboard settings do not work so you can remove those, since everything can be controlled using the brightness button anyways. 

Here is the log file.
[log.txt](https://github.com/user-attachments/files/18914725/log.txt)

**Reply by seerge** [reply]
@mehtabmahir did you try restarting G-Helper after stopping services ? Do hotkeys work after that ?

What specific keyboard settings are you talking about ?

**Reply by mehtabmahir** [reply]
@seerge Sorry I didn't mean the hotkey didn't work, that part does in fact work. I meant a feature of the laptop where the keyboard backlight turns off when you're in a bright environment, that stops working after stopping the services, so maybe that one should be excluded. Restarting G Helper did not fix it.

The specific keyboard settings that do not work are:
- Animations
- Color (of course since it's not RGB)
- Animation Speed

What works:
- Timeout Speed

Also, I realized G Helper is the thing that is causing it to get power limited at 24 watts. I believe it is because G Helper has 3 performance modes, but in MyASUS there are actually 4: Whisper, Standard, Performance, and Full-Speed Mode. So, G Helpers "Turbo" setting actually enables the "Performance" mode instead of the "Full-Speed" mode. After closing G Helper and enabling Full-Speed mode from MyASUS, it actually does stay steady at 28 watts instead of 24 watts.

**Reply by seerge** [reply]
@mehtabmahir 

That must be some other "proprietary" feature that Zenbook S16 have :) It could be some **light sensor** or so. If you want to contribute, we can try to reverse-engineer that, but it's difficult. Is it Asus Optimization service that handles it ?


For `Full-Speed` mode - I was answering this already in couple of other issues https://github.com/seerge/g-helper/issues/3194 https://github.com/seerge/g-helper/issues/2589

You can set `Full Speed` mode already but you need to add it config first :

1. Create a new custom mode based on `Turbo` (by pressing + under `Fans+Power`)
2. Quit app
3. Open `%appdata%\GHelper\config.json` and set following parameter to `3` (it will be `2` there) `"mode_base_3" : 3, `
4. Save config, and start app.

**Reply by mehtabmahir** [reply]
@seerge Hello, thank you for your reply. Yes, it does seem to be the ASUS Optimization service. After I disable it in services.msc, the feature does stop working. For now, I can stop ASUS services from G Helper and then enable that one service manually. I think we can exclude that service for this specific laptop. (EDIT: just realized that service triggers MyASUS to open so nevermind.) I definitely want to contribute in any way I can, to make g helper work well with this device. If you can point me in the right direction, I can try coding in the changes myself.

I've followed those steps, and it seemed to have worked perfectly, except "mode_base_3" was originally set to 1 instead of 2. Regardless, its working great now, thank you!

**Reply by mehtabmahir** [reply]
@seerge I saw one of your latest changes for hiding the gpu section, I'd love to try it out if you have the build

**Reply by seerge** [reply]
@mehtabmahir it's already live as 0.205. I have moved second/GPU fan metrics to Screen row. It usually has third fan readings there, but i suspect there are no devices that would have no GPU and 3 fans at the same time.

**Reply by mehtabmahir** [reply]
@seerge Just saw the pre-release build, didn't notice it before. But it seems to be working, just the gpu fan shows up as 0 rpm all the time now but its not a big deal. I think it is because you thought the gpu fan line wouldn't show up but it still does, I've attached a screenshot. 

I think it was something to do with this line: labelMidFan.Text = "AUX " + HardwareControl.gpuFan;

Thank you for the great work you're doing! 

![image](https://github.com/user-attachments/assets/e5e8e650-90fd-460b-b8f2-a47359ccf976)

**Reply by seerge** [reply]
@mehtabmahir hm, but do you have 2 fans ? or one ? cause if previous version shows correct fan speed - this one should also.

**Reply by mehtabmahir** [reply]
@seerge Yes you're 100% right, I just double checked and it was from a lack of load. It is working perfectly. Thanks!!

**Reply by mehtabmahir** [reply]
@seerge Hope this helps, it is already very usable and I've already replaced MyASUS with this so thank you so much for your work.

Zenbook S16 Ryzen 9 365

What's currently working:
- Performance Modes (Have to use custom profile for full peformance)
- Refresh Rates
- Flicker Free Dimming
- Color Gamut (Native, DCIP3, DisplayP3, sRGB)
- Key Bindings (only fn-f12)
- Laptop Backlight slider
- Backlight timeout

What's currently not working:
- Auto brightness turning on/off backlight (relies on ASUS Optimization service)
- Visual Mode (Default, Vivid, Manual, Eyecare) but it works when pressing fn+C which is odd
- Color temperature
- Keyboard animations/color (keyboard has no RGB, but has a breathing animation (only happens during bootup) I wish I could use)
- fnlock should be fn+esc, visual mode should be fn+C (these don't seem to change)
- Animaton Speed

**Reply by seerge** [reply]
@mehtabmahir 

> Visual Mode (Default, Vivid, Manual, Eyecare) but it works when pressing fn+C which is odd

Does FN+C work w/o G-Helper running ? If so - they could be handled by hardware at all. How do you change them via asus apps ? Is it MyASUS or what ?

> Keyboard animations/color (keyboard has no RGB, but has a breathing animation (only happens during bootup) I wish I could use)

If your device does not support breathing animation while awake, it can't be added. Do I understand correctly that "Static" is the only mode it supports ?

> fnlock should be fn+esc, visual mode should be fn+C (these don't seem to change)

FN+ESC is a hardware toggle for FN-Lock on devices that support it from the beginning. If it works already - then good.

**Reply by skbtwiz** [reply]
I'll jump in to answer these questions. 

1. Fn+C only toggles visual modes `when G-Helper is running`, even with all Asus services stopped. Additionally, when I change the Fn+C keybind in the Extra settings menu, nothing changes. However when I change the Fn+V keybind from Visual Mode to Mute for example, Fn+C changes to mute instead. Basically, Fn+C is actually following what is set under "Fn+V" in the settings, instead of doing what is set under "Fn+C". The actual "Fn+C" setting does not seem to change anything.

2. Normally Visual Modes are changed using the MyAsus app. 

3. As far as keyboard animations are concerned, I'm pretty sure the device only supports static mode in single color when awake.

Not to pile on but I would humbly like to bring some additional items to your notice. 

1. The boot sound toggle in the Extra menu doesn't seem to do anything - is this a setting that is supposed to be a part of all Asus laptops (in which case the toggle is broken) or can some laptops simply not have the ability to enable this sound (in which case the toggle is working as intended)?

2. Currently I use the presets functionality in MyAsus app to change CPU mode, display framerate, Visual mode, and the Sound mode (Dolby Atmos presets) altogether when switching between productivity, media consumption, and gaming (these are my three presets). While creating a global presets functionality could be a more intense undertaking for the future, in the meanwhile is there some way to figure out how MyAsus is adjusting Dolby Atmos modes even with the Dolby Atmos app not running? If so, I can perhaps try my hand at working on a PR for adding a Dolby Atmos mode-switcher with your permission.

I just got this laptop and would love to contribute in any way possible to this wonderful project!

**Reply by mehtabmahir** [reply]
@seerge Sorry that I just saw your message but as the other user said, fn + C does seem to work without ASUS services, and G Helper updates to the appropriate settings when using the hotkey as well.

Yes static does seem to be the only mode supported but the breathing animation stops a few seconds after logging in, so maybe there’s a small possibility to use it? If not, it’s no big deal.

As for @skbtwiz ’s concerns:

You’re right, this laptop doesn’t have a boot sound but some other laptops like the G14 does have one. So it would be best if this setting is entirely removed for our zenbook. 

As for your second concern, I have G helper automating the modes for me, when plugged in I sometimes use the Custom profile mentioned above to get the full 28 watt sustained performance, and have it switching to silent mode on battery. I don’t think there’s a way to tie all those settings together currently, but you can of course try to implement it. I prefer a set it and forget it preset for Dolby, and use the default mode for the display. If you want I can send you my settings for Dolby, it makes the sound way better.

**Reply by thureos** [reply]
> it works great at 40 watts but once it thermal throttles

Hey @mehtabmahir could you please run timespy with 40w and full fan speed and post the results? I'd greatly appreciate it

**Reply by mehtabmahir** [reply]
@thureos When it starts it uses a little over 40 watts then slowly goes down (over a few minutes) to 28 watts and stays there.


![image](https://github.com/user-attachments/assets/fc29fc89-7944-4ab7-84c9-f8327351d9dc)

**Reply by mehtabmahir** [reply]
This is without adjusting the power limit at all so I'll stick to that.

![image](https://github.com/user-attachments/assets/728e3ded-72a7-4c77-aab2-0a73a997475a)

**Reply by thureos** [reply]
Thank you so much! I got this with mine using UXTU and G-helper and full fan power. Sadly it never moved from 28W
![image](https://github.com/user-attachments/assets/b764c5e5-6bb2-4cd7-9f14-4dbf8f986d70)

**Reply by mehtabmahir** [reply]
@thureos  can you try full speed mode on MyASUS without any g helper, and restart just to be sure? The second score I gave was without any third party software. It initially boosted to 45 and slowly dropped to 28 over the course of 3-5 mins

**Reply by thureos** [reply]
@mehtabmahir Will do this evening. also remember I have the 370 and it appears you have the 365, don't know if that affects a lot

**Reply by mehtabmahir** [reply]
@thureos Yeah, I noticed and the difference isn’t as big as I expected. If you can check the wattage it is using, that would be great too. The built in Radeon overlay shows it so I’d use that. Thanks!

**Comment by thureos** [comment]
https://www.3dmark.com/3dm/128199423

8828. Sad
Watts never hit +28

**Reply by mehtabmahir** [reply]
9070 xt? lol

**Comment by thureos** [comment]
Yeah, I'm running a eGpu
